src/pages/signinBlockstack.js

```diff
@@ -0,0 +1,259 @@
+    }).then(checkResponseStatus);
+  }
+
+  signinEmailPost(user, redirect) {
+    const websiteUrl = getWebsiteUrl();
+    return fetch('/api/users/signin', {
+      method: 'POST',
+      headers: {
+        ...this.addAuthTokenToHeader(),
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ user, redirect, websiteUrl }),
+    }).then(checkResponseStatus);
+  }
+
+  addAuthTokenToHeader(obj = {}) {
+    const accessToken = localStorage.getItem('accessToken');
+    if (!accessToken) return obj;
+    return {
+      Authorization: `Bearer ${accessToken}`,
+      ...obj,
+    };
+  }
+
+  render() {
+    const { userData, exists, existsEmail, loading, error } = this.state;
+    return (
+      <Page title="Blockstack Login Successful">
+        <Container pt={4} pb={6} px={2} background="linear-gradient(180deg, #EBF4FF, #FFFFFF)" textAlign="center">
+          <Flex justifyContent="center" mb={4}>
+            <Icon size="60" />
+          </Flex>
+          <H3 as="h1" fontWeight="800">
+            {userData && <>{userData.username}</>}
+            {loading && (
+              <>
+                <br />
+                ... stacking blocks ...
+              </>
+            )}
+            {!loading && exists && (
+              <>
+                <br />
+                You have already a profile. Signing in ...
+              </>
+            )}
+            {!loading && !exists && existsEmail && (
+              <>
+                <br />
+                You have already a profile.
+                <br />
+              </>
+            )}
+            {!loading && !exists && !existsEmail && (
+              <>
+                <br />
+                ... creating your profile ...
+              </>
+            )}
+          </H3>
+          {!loading && !exists && existsEmail && (
+            <>
+              <br />
+              Please sign in with the magic link to connect your blockstack id to your profile.
+              <br />
+              <H3 as="h1" fontWeight="800">
+                Your magic link is on its way!
+              </H3>
+              <P fontSize="LeadParagraph" lineHeight="LeadParagraph" color="black.900" mt={4}>
+                We&apos;ve sent it to <strong>{userData.email}</strong>.
+              </P>
+            </>
+          )}
+          {error && <Error message={error} />}
+        </Container>
+      </Page>
+    );
+  }
+}
+
+SignInBlockstack.propTypes = {
+  authResponse: PropTypes.string,
+  next: PropTypes.string,
+  organizationData: PropTypes.object,
+  createUser: PropTypes.func,
+  getLoggedInUser: PropTypes.func.isRequired, // from withLoggedInUser
+};
+
+const addCreateUserMutation = graphql(createUserByPublicKeyQuery, {
+  props: ({ mutate }) => ({
+    createUser: variables => mutate({ variables }),
+  }),
+});
+
+export default addCreateUserMutation(withData(withLoggedInUser(SignInBlockstack)));
```

Comment:
This just changed recently in https://github.com/opencollective/opencollective-frontend/pull/1954

`withLoggedInUser` is being deprecated and needs to be replaced by `withUser`, check the work in the related PR and you will get a clear idea.

Reply:
No problem. Replaced with `withUser` now.